packages/utils/src/abi_encoder/evm_data_types/method.ts

```diff
@@ -44,7 +44,20 @@ export class MethodDataType extends AbstractSetDataType {
         return returnValues;
     }
 
-    public getSignature(): string {
+    public strictDecodeReturnValue<T>(returndata: string, rules?: DecodingRules): T {
```

Comment:
Is there a point in having a non-strict version of this still? I feel like we could probably add generic types to the regular `decodeReturnValues`. 

Also this doesn't have to be in this PR if it's a lot of changes, but we should add generic types to the regular `decode` function as well.

Reply:
I've created a task to use generics for all decoding. Shouldn't take long but I think it makes sense to have in a separate PR as well.